protocol/proofs/anchor-state-poison-protection.md

```diff
@@ -0,0 +1,121 @@
+# AnchorStateRegistry Poisoning Mitigation
+
+## Purpose
+
+Provides a design for an updated version of the `AnchorStateRegistry` with a clearer role within
+the system as a whole that is not subject to the "poisoning" problem found in the current design.
+
+## Context
+
+The `FaultDisputeGame` contract is responsible for managing the process of disputing a fault in the
+computation of a state transition function. The `FaultDisputeGame` runs the state transition
+function from some starting state that is known to be valid. We refer to this starting state as the
+"anchor state". In the context of the OP Stack, the anchor state will originally either be the
+genesis state or the state of the system at some finalized L2 block height that is considered to be
+correct by social consensus.
+
+`FaultDisputeGame` contracts pull the current anchor state from the `AnchorStateRegistry` contract.
+As the system runs, the anchor state is continuously updated to reflect the latest finalized state
+as viewed by the proof system at any given time. `FaultDisputeGame` instances that resolve in favor
+of the defender will trigger the `AnchorStateRegistry` to update the anchor state to the newly
+finalized state.
+
+## Problem Statement
+
+The `AnchorStateRegistry` can become "poisoned" because the ultimate view of state validity
+currently actually lives inside of the `OptimismPortal` contract. An anchor state is considered
+"poisoned" if the `OptimismPortal` contract has blacklisted the dispute game that caused that
```

Comment:
Newbie Questions about the current design: 

> The Ultimate view of the state validity lives into the OptimismPortal contract.

- [ ] does  the anchor can be poisoned if the game is blacklisted **before** resolved? Does the game will still `resolve()` and write into the `AnchorStateRegistry` if blacklisted? If yes, does there is a specific reason that allowing blacklisted game to resolve? More generally, It that possible to have a fonction callable by the guardian that is `BlacklistGameAndStopGame` exactly like the previous one but would stop a malicious game for being resolved. 

Edit: there is only 3 status for the current `FaultDisputeGame`

```solidity
enum GameStatus {
    // The game is currently in progress, and has not been resolved.
    IN_PROGRESS,
    // The game has concluded, and the `rootClaim` was challenged successfully.
    CHALLENGER_WINS,
    // The game has concluded, and the `rootClaim` could not be contested.
    DEFENDER_WINS
}
```

Reply:
So we generally want to create a distinction between the underlying games and a subjective view of validity on top of those games. This is why we allow dispute games to resolve, blacklisting is a subjective thing that the `OptimismPortal` does on top of games. Poisoning happens because the `OptimismPortal` blacklists a game but the `AnchorStateRegistry` is not aware of this and allows new games to be created on top of blacklisted games, which then leads to a situation where the `OptimismPortal` should logically also consider these new games to be blacklisted.

Reply:
What we want to do in this proposal is make it no longer possible for new games to be created from blacklisted states so that this implicit invalidity doesn't happen anymore.